### DISCUSSION TITLE
Token selection logic is not good

### DISCUSSION THREAD

**Discussion by doronbehar** [goreleaser/goreleaser#2732] [Ideas, Issue Triage, and General discussions]
For the first time I ran `goreleaser`, I encountered this very unclear error message:

```
   • building...
   • loading config file       file=.goreleaser.yml
   • loading environment variables
   ⨯ build failed after 0.00s error=multiple tokens defined. Only one is allowed
```

And I didn't understand what's wrong with my configuration file. Apparently, after reading the source code I read this:

https://github.com/goreleaser/goreleaser/blob/main/internal/pipe/env/env.go#L81-L104

And I realized that I have to unset one of my `GIT{LA,HU}B_TOKEN` environment variables, which are both set by default in my shell rc files.

I think the error message could at least be improved, or you can default to the GitHub token?

**Comment by caarlos0** [comment]
should be better after https://github.com/goreleaser/goreleaser/pull/2733

Thanks!
